src/main/java/com/icosillion/podengine/models/Podcast.java

```diff
@@ -67,7 +69,7 @@ public Podcast(URL feed) throws InvalidFeedException, MalformedFeedException {
         } catch (DocumentException e) {
             throw new InvalidFeedException("Error parsing feed XML.", e);
         } finally {
-            IOUtils.closeQuietly(ubis);
+            IOUtils.closeQuietly(bomInputStream);
```

Comment:
I'm reading this fairly cursorily, but I see that this changes the argument to `closeQuietly` from the object Dan created to an object created by the upstream. Is that intentional?

Also, I don't know what Byte Order Marks are, but it seems like Dan introduced some code to deal with them some time ago, and that later the upstream author added code to strip BOM. Is it possible this mashed-up code is trying to deal with the same problem in two different ways, and that we can just go with the upstream approach to ease the maintenance burden?

Reply:
Oh I think I see what's happened here. You _did_ favor the upstream approach to BOM, but you didn't completely remove Dan's code.